### PR TITLE
Remove sidebar upsell nudges from Calypso

### LINKED ISSUE #58613: [Bug]: Possible for calypso to display multiple upsell nudges in sidebar
### Quick summary

Calypso may display multiple upsell nudges in the sidebar - one from the server JITMs and another originating from Calypso. This seems over-powering.

#52368 will eventually obsolete this issue, it's just that this issue means it'll take a little bit longer to deploy bits of that one.

### Steps to reproduce

1. Make site eligible for a server-originated JITM
    * The easiest way to do this is to open jitm-engine.php in your wpcom sandbox and change the eligibility rules & feature_classes for messages that are defined using `with_template( 'sidebar-banner' );`
2. Ensure site is eligible for a calypso-originated upsell notice
    * The easiest way is to have a site eligible for a free domain but hasn't taken one i.e. it's on an annual plan and a .wordpress.com subdomain
3. Access the site admin using calypso

### What you expected to happen

The site to display just the server-originated JITM

### What actually happened

The site displayed both messages

image

(please ignore content of the upsell nudges, this pairing shouldn't happen outside of testing)

### Context

Found while working on #52368 which aims to remove the calypso upsells

### Browser

_No response_

### Simple/Atomic

Simple, Atomic

### Other notes

There is some [code in calypso to prevent both from being displayed](https://github.com/Automattic/wp-calypso/blob/4265ac7786b2331f5c89656650f9b3bd7e3020a6/client/my-sites/current-site/notice.jsx#L286) but it doesn't work. It relies upon the SiteNotice component having a hasJITM property that is true when a server-originated JITM exists, however this property is never set so it's always null. It's child from calypso/blocks/jitm is the one that gets a JITM and displays it.

### Reproducibility

Consistent

### Severity

_No response_

### Available workarounds?

_No response_

### Workaround details

I don't think that this actually happens with the current created JITMs, but this is just because of the rules they have at present, which could easily change. In #52368 the calypso originated upsells will be removed completely by the end

### PR DESCRIPTION
**Pull request by dsas**
#### DO NOT MERGE
Requires D69690-code to be merged first.

---

Remove sidebar upsell nudges from calypso as D69690-code moves them into
the JITM infrastructure. This means that the nudges will appear in
wp-admin as well as calypso.

#### Testing instructions

1. Apply D69690-code to your sandbox and follow testing instructions there
2. Check you don't see the same message twice on the same screen, i.e. that #58613 is fixed

Related to #52368

### CODE REVIEW

**Review by mmtr — APPROVED** [review]
LGTM. Let's just hold off merging this until D69690-code lands first.

File: client/my-sites/current-site/notice.jsx [hunk 1]
```diff
@@ -245,14 +112,13 @@ export class SiteNotice extends Component {
 	}
 
 	render() {
-		const { site, isMigrationInProgress, hasJITM } = this.props;
+		const { site, isMigrationInProgress } = this.props;
```
Comment (mmtr) [inline client/my-sites/current-site/notice.jsx:115]: `hasJITM` is no longer used in this component now, so I guess we can remove the line below as well:

https://github.com/Automattic/wp-calypso/blob/76e7c58d5385863f9c1a93469bc737e673c313fa/client/my-sites/current-site/notice.jsx#L170

Comment (dsas) [reply client/my-sites/current-site/notice.jsx:115]: will tidy this up as a separate PR rather than re-open and delay this one

**Comment by jsnajdr** [comment]
Hi @dsas :wave: You can remove a lot of additional code from the `SiteNotice` component. None of the connected `clickDomain*` actions are used any more, and neither are the `hasDomainCredit`, `isPlanOwner` and `domainUpsellNudgeDismissedDate` props. Would you mind doing a follow-up finishing the removal?

**Comment by dsas** [comment]
Yep, definitely will @jsnajdr thanks for the pointers.

**Comment by dsas** [comment]
Opened #59058 to deal with the unused code @mmtr and @jsnajdr mentioned.

**Pull request merged**